api/lql.go

```diff
@@ -18,24 +18,201 @@
 
 package api
 
+import (
+	"encoding/json"
+	"net/url"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	reLQL                  string = `(?ms)^(\w+)\([^)]+\)\s*{`
+	LQLQueryTranslateError string = "unable to translate query blob"
+)
+
+type LQLQuery struct {
+	ID             string `json:"LQL_ID,omitempty"`
+	StartTimeRange string `json:"START_TIME_RANGE,omitempty"`
+	EndTimeRange   string `json:"END_TIME_RANGE,omitempty"`
+	QueryText      string `json:"QUERY_TEXT"`
+	// QueryBlob is a special string that supports type conversion
+	// back and forth from LQL to JSON
+	QueryBlob string `json:"-"`
+}
+
+func (q *LQLQuery) Validate(allowEmptyTimes bool) error {
+	// translate
+	if err := q.Translate(); err != nil {
+		return err
+	}
+	// validate range
+	if err := q.ValidateRange(allowEmptyTimes); err != nil {
+		return err
+	}
+	// validate query
+	if q.QueryText == "" {
+		return errors.New("query should not be empty")
+	}
+	return nil
+}
+
+func (q *LQLQuery) Translate() error {
+	// query
+	if err := q.TranslateQuery(); err != nil {
+		return err
+	}
+	// start
+	if start, err := q.TranslateTime(q.StartTimeRange); err != nil {
+		return err
+	} else {
+		q.StartTimeRange = start
+	}
```

Comment:
This else is not needed since you are returning on line `69`
```suggestion
	}
	q.StartTimeRange = start
```

Reply:
`else` is needed because `start` is assigned via `if`.